modules/querier/external/cloud_run.go

```diff
@@ -0,0 +1,19 @@
+package external
+
+import (
+	"context"
+	"fmt"
+
+	"golang.org/x/oauth2"
+	"google.golang.org/api/idtoken"
+)
+
+// Return Oauth2 token according to our google application credentials:
+// https://cloud.google.com/run/docs/authenticating/service-to-service#acquire-token
+func googleToken(ctx context.Context, endpoint string) (*oauth2.Token, error) {
+	ts, err := idtoken.NewTokenSource(ctx, endpoint)
```

Comment:
this will be called 100s or 1000s of times/second. what is the correct way to use `NewTokenSource`? should we cache it locally or is it meant to be invoked like this?

Reply:
Great catch! Yes, we should cache it. I pushed up an implementation which leverages a cached `oauth2.TokenSource` per endpoint.